web/impact/impact/v1/views/user_list_view.py

```diff
@@ -114,38 +113,20 @@ def _invalid_keys(self, user_keys):
         for key in set(user_keys) - set(UserHelper.INPUT_KEYS):
             self.errors.append(INVALID_KEY_ERROR.format(key))
 
-    def _results(self, limit, offset):
-        queryset = User.objects.all()
-        updated_at_after = self.request.query_params.get(
-            'updated_at.after', None)
-        updated_at_before = self.request.query_params.get(
-            'updated_at.before', None)
-        if updated_at_after or updated_at_before:
-            queryset = _filter_profiles_by_date(
-                queryset,
-                updated_at_after,
-                updated_at_before)
-        count = queryset.count()
-        return (count,
-                [UserHelper(user).serialize()
-                 for user in queryset[offset:offset + limit]])
-
-
-def _filter_profiles_by_date(queryset, updated_at_after, updated_at_before):
-    updated_at_after = parse_date(updated_at_after)
-    updated_at_before = parse_date(updated_at_before)
-    if updated_at_after:
-        queryset = queryset.filter(
-            Q(expertprofile__updated_at__gte=updated_at_after) |
-            Q(entrepreneurprofile__updated_at__gte=updated_at_after) |
-            Q(memberprofile__updated_at__gte=updated_at_after))
-    if updated_at_before:
-        queryset = queryset.exclude(
-            Q(expertprofile__updated_at__gt=updated_at_before) |
-            Q(entrepreneurprofile__updated_at__gt=updated_at_before) |
-            Q(memberprofile__updated_at__gt=updated_at_before)
-        )
-    return queryset
+    def _filter_by_date(self, qs, after, before):
```

Comment:
Should I point out that this function is > 10 lines?

Reply:
Depends on how you count lines and CodeClimate gives it a pass.  This is ugly code that is going to get refactored once we clean up the profile handling.